xbmc/media/decoderfilter/DecoderFilterManager.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Copyright (C) 2013-2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+
+ /**
+ * \file media\hwdecoder\DecoderFilterManager.cpp
+ * \brief Implements CDecoderFilterManager class.
+ *
+ */
+
+#include "DecoderFilterManager.h"
+#include "cores/VideoPlayer/DVDStreamInfo.h"
+
+#include "filesystem/File.h"
+#include "threads/SingleLock.h"
+#include "utils/log.h"
+#include "utils/XMLUtils.h"
+#include "Util.h"
+
+static const char* TAG_ROOT = "decoderfilter";
```

Comment:
What's the reason for using static members and `const char*` instead of class members and `std::string`?

Reply:
What are the reasons for declaring them in class as they are not meant to be exposed.

Reply:
can they not be declared `private` then?

Reply:
I (currently) don't see any reason for having them in the header file. But I'm open for changes if they make sense. But I want to understand them before I change.

Reply:
You can still initialize them in the header file and you can declare them private to the class. As it is right now they are global and not even in an anonymous namespace.

Reply:
IMO having (implementation-private) constants in the implementation file is fine. `static` is functionally equivalent to an anonymous namespace, but we usually prefer the latter.

Reply:
Ah yea, they're `const`, not sure what I was thinking this morning. Sorry for the noise.